code/datums/diseases/advance/advance.dm

```diff
@@ -126,6 +126,26 @@ GLOBAL_LIST_INIT(advance_cures, list(
 			return 1
 	return 0
 
+/datum/disease/advance/proc/GenerateSymptomsBySeverity(sev_min, sev_max, amount = 1)
+
+	var/list/generated = list() // Symptoms we generated.
+
+	var/list/possible_symptoms = list()
+	for(var/symp in GLOB.list_symptoms)
+		var/datum/symptom/S = new symp
+		if(S.severity >= sev_min && S.severity <= sev_max)
+			if(!HasSymptom(S))
+				possible_symptoms += S
+
+	if(!possible_symptoms.len)
+		return generated
+
+	for(var/i = 1; amount >= i && possible_symptoms.len; i++)
+		generated += pick_n_take(possible_symptoms)
```

Comment:
You don't need to check `length(possible_symptoms)` here since you've already returned early. Plus, DM has some nicer syntax for this.
```suggestion
	for(var/i in 1 to amount)
		generated += pick_n_take(possible_symptoms)
```